packages/components/nodes/chatmodels/ChatGoogleGenerativeAI/ChatGoogleGenerativeAI.ts

```diff
@@ -0,0 +1,108 @@
+    description: string
+    baseClasses: string[]
+    credential: INodeParams
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'ChatGoogleGenerativeAI'
+        this.name = 'chatGoogleGenerativeAI'
+        this.version = 2.0
+        this.type = 'ChatGoogleGenerativeAI'
+        this.icon = 'gemini.png'
+        this.category = 'Chat Models'
+        this.description = 'Wrapper around Google Gemini large language models that use the Chat endpoint'
+        this.baseClasses = [this.type, ...getBaseClasses(ChatGoogleGenerativeAI)]
+        this.credential = {
+            label: 'Connect Credential',
+            name: 'credential',
+            type: 'credential',
+            credentialNames: ['googleGenerativeAI'],
+            optional: true,
+            description: 'Google Generative AI credential.'
+        }
+        this.inputs = [
+            {
+                label: 'Cache',
+                name: 'cache',
+                type: 'BaseCache',
+                optional: true
+            },
+            {
+                label: 'Model Name',
+                name: 'modelName',
+                type: 'options',
+                options: [
+                    {
+                        label: 'gemini-pro',
+                        name: 'gemini-pro'
+                    }
+                ],
+                default: 'gemini-pro',
+                optional: true
+            },
+            {
+                label: 'Temperature',
+                name: 'temperature',
+                type: 'number',
+                step: 0.1,
+                default: 0.9,
+                optional: true
+            },
+            {
+                label: 'Max Output Tokens',
+                name: 'maxOutputTokens',
+                type: 'number',
+                step: 1,
+                optional: true,
+                additionalParams: true
+            },
+            {
+                label: 'Top Probability',
+                name: 'topP',
+                type: 'number',
+                step: 0.1,
+                optional: true,
+                additionalParams: true
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
+        const credentialData = await getCredentialData(nodeData.credential ?? '', options)
+        const apiKey = getCredentialParam('googleGenerativeAPIKey', credentialData, nodeData)
+
+        const temperature = nodeData.inputs?.temperature as string
+        const modelName = nodeData.inputs?.modelName as string
+        const maxOutputTokens = nodeData.inputs?.maxOutputTokens as string
+        const topP = nodeData.inputs?.topP as string
+        const cache = nodeData.inputs?.cache as BaseCache
+
+        const obj = {
+            apiKey: apiKey,
+            modelName: modelName,
+            maxOutputTokens: 2048
+        }
+
+        if (maxOutputTokens) obj.maxOutputTokens = parseInt(maxOutputTokens, 10)
```

Comment:
think we can standardize and use `parseFloat`

Reply:
Tokens are usually parseInt. But changed temperature to parseFloat and pushed the changes.